### ISSUE TITLE
Respect parameter format in Bind

### ISSUE THREAD

**Issue by sunng87** [sunng87/pgwire#45]
The JDBC client encodes text (oid: 1043) value as text format. In our `Portal` API, format is ignored at the moment, we will at least check the format and attempt parse parameter value using known types

```
Frame 8268: 178 bytes on wire (1424 bits), 178 bytes captured (1424 bits) on interface lo, id 0
Ethernet II, Src: 00:00:00_00:00:00 (00:00:00:00:00:00), Dst: 00:00:00_00:00:00 (00:00:00:00:00:00)
Internet Protocol Version 4, Src: 127.0.0.1, Dst: 127.0.0.1
Transmission Control Protocol, Src Port: 52608, Dst Port: 5432, Seq: 164, Ack: 145, Len: 112
PostgreSQL
    Type: Parse
    Length: 51
    Statement: 
    Query: INSERT INTO mytable VALUES ($1, $2)
    Parameters: 2
        Type OID: 20
        Type OID: 1043
PostgreSQL
    Type: Bind
    Length: 37
    Portal: 
    Statement: 
    Parameter formats: 2
        Format: Binary (1)
        Format: Text (0)
    Parameter values: 2
        Column length: 8
        Data: 0000000000000002
        Column length: 5
        Data: 6a65727279
    Result formats: 0
PostgreSQL
    Type: Describe
    Length: 6
    Portal: 
PostgreSQL
    Type: Execute
    Length: 9
    Portal: 
    Returns: all rows
PostgreSQL
    Type: Sync
    Length: 4
```

**Comment by sunng87** [comment]
After #50 , we allows developer to access Bind data and both data encoder apis for both text and binary format. So it's up to developer to follow the format. However, this API is not perfect and better automate this.

**Comment by sunng87** [comment]
Resolved in #66. We now allows user to set formats for `DataRowEncoder`
